src/java/org/apache/cassandra/db/filter/RowFilter.java

```diff
@@ -529,7 +530,7 @@ protected ByteBuffer getValue(TableMetadata metadata, DecoratedKey partitionKey,
                     return row.clustering().bufferAt(column.position());
                 default:
                     Cell<?> cell = row.getCell(column);
-                    return cell == null ? null : cell.buffer();
+                    return Cell.getValidCellBuffer(cell, nowInSeconds());
```

Comment:
sorry, been bad about getting back to this... im not a fan of `nowInSeconds`  being called here as we should really do this once per query... im trying to look at `org.apache.cassandra.db.ReadCommand#executeLocally` to see if I can see why this wasn't already filtered out...

Reply:
`org.apache.cassandra.db.filter.RowFilter#filter(org.apache.cassandra.schema.TableMetadata, long)`

we are passed in a `nowInSeconds`!  It looks like we can refactor the caller to pass this value in, that way we don't call `nowInSeconds` for every cell (this can become a problem)

Reply:
once we plumb `nowInSeconds` from the caller through here, then I am +1 to this patch